addon/v1alpha1/types_managedclusteraddon.go

```diff
@@ -0,0 +1,131 @@
+	// name of the referent.
+	// +kubebuilder:validation:Required
+	// +required
+	Name string `json:"name"`
+}
+
+// AddOnStatusCondition represents the state of the add-on's
+// managed and monitored components.
+// +k8s:deepcopy-gen=true
+type AddOnStatusCondition struct {
+	// type specifies the aspect reported by this condition.
+	// +kubebuilder:validation:Required
+	// +required
+	Type AddOnStatusConditionType `json:"type"`
+
+	// status of the condition, one of True, False, Unknown.
+	// +kubebuilder:validation:Required
+	// +required
+	Status metav1.ConditionStatus `json:"status"`
+
+	// lastTransitionTime is the time of the last update to the current status property.
+	// +kubebuilder:validation:Required
+	// +required
+	LastTransitionTime metav1.Time `json:"lastTransitionTime"`
+
+	// reason is the CamelCase reason for the condition's current status.
+	// +optional
+	Reason string `json:"reason,omitempty"`
+
+	// message provides additional information about the current condition.
+	// This is only to be consumed by humans.
+	// +optional
+	Message string `json:"message,omitempty"`
+}
+
+// AddOnStatusConditionType is an aspect of agent state.
+type AddOnStatusConditionType string
+
+const (
+	// Available indicates that the agent is functional and available in the cluster.
+	Available AddOnStatusConditionType = "Available"
+
+	// Progressing indicates that the operator is actively rolling out new code,
+	// propagating config changes, or otherwise moving from one steady state to
+	// another.  Operators should not report progressing when they are reconciling
+	// a previously known state.
+	Progressing AddOnStatusConditionType = "Progressing"
+
+	// Degraded indicates that the operator's current state does not match its
+	// desired state over a period of time resulting in a lower quality of service.
+	// The period of time may vary by component, but a Degraded state represents
+	// persistent observation of a condition.  As a result, a component should not
+	// oscillate in and out of a Degraded state.  A service may be Available even
+	// if it's degraded.  For example, your service may desire 3 running pods, but 1
+	// pod is crash-looping.  The service is Available but Degraded because it
+	// may have a lower quality of service.  A component may be Progressing but
+	// not Degraded because the transition from one state to another does not
+	// persist over a long enough period to report Degraded.  A service should not
+	// report Degraded during the course of a normal upgrade.  A service may report
+	// Degraded in response to a persistent infrastructure failure that requires
+	// administrator intervention.  For example, if a control plane host is unhealthy
+	// and must be replaced.  An operator should report Degraded if unexpected
+	// errors occur over a period, but the expectation is that all unexpected errors
+	// are handled as operators mature.
+	Degraded AddOnStatusConditionType = "Degraded"
```

Comment:
Break this out a bit and highlight with code or quotes the values of that status. Also, please consider a definition list:

1. Available: Indicates......
(Make a list like this -- easier to see definitions.)
2. But a `Degraded` state...

(Within a paragraph, set it apart with code blocks.)

Reply:
this section is copy and pasted from https://github.com/openshift/api/blob/master/config/v1/types_cluster_operator.go#L138-L175